### PR TITLE
[AUTOMATE-1607] Move tests in flaky system authz tests to better homes

### PR DESCRIPTION
**Pull request by tylercloke**
### :nut_and_bolt: Description: What code changed, and why?

Moved all the system tests we've been struggling with to either the OPA engine tests or cypress API level tests. In a few instances, we already had coverage in the OPA tests anyway.

Also removed all the integer leading parts of the file names from the cypress tests (01_..., ect).

### :athletic_shoe: How to Build and Test the Change

Run :allthethings: tests.

### :white_check_mark: Checklist

- [x] I have read the [CONTRIBUTING document](https://github.com/chef/automate/blob/master/CONTRIBUTING.md).
- [x] Tests added/updated?
- [ ] Docs added/updated?
- [x] All commits have been signed-off for the [Developer Certification of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

### CODE REVIEW

**Comment by blakestier** [comment]
This seems cool . . . but I haven't checked out the failing tests. Maybe has something to do with cleanup/order of tests?

File: e2e/cypress/integration/api/iam/project_assignment.spec.ts [hunk 1]
```diff
@@ -0,0 +1,452 @@
...
+                name: testObjectID,
+                projects: [authorizedProject.id, unauthorizedProject.id]
+            },
+            failOnStatusCode: false
+        }).then((resp) => {
+            expect(resp.status).to.equal(403);
+        });
+    });
+
+    it('creating an object without projects is not allowed ' +
```
Comment (bcmdarroch) [inline e2e/cypress/integration/api/iam/project_assignment.spec.ts]: I think this case can be moved to iam_integration

File: e2e/cypress/integration/api/iam/project_assignment.spec.ts [hunk 2]
```diff
@@ -0,0 +1,452 @@
...
+            },
+            method: 'POST',
+            url: '/apis/iam/v2beta/teams',
+            failOnStatusCode: false
+        }).then((resp) => {
+            expect(resp.status).to.equal(403);
+        });
+    });
+
+    it('creating an object without projects is allowed ' +
```
Comment (bcmdarroch) [inline e2e/cypress/integration/api/iam/project_assignment.spec.ts]: another case that could be added to iam_integration

File: components/authz-service/server/v2/system_integration_test.go [hunk 3]
```diff
@@ -1,780 +0,0 @@
...
-			Values:    values3,
-			Operator:  api_v2.ProjectRuleConditionOperators_MEMBER_OF,
-		},
-	}
-
-	cases := []struct {
-		desc string
-		f    func(*testing.T)
-	}{
-		{"when no rules exist, returns an empty list", func(t *testing.T) {
```
Comment (bcmdarroch) [inline components/authz-service/server/v2/system_integration_test.go]: did these tests already move to Cypress?

File: e2e/cypress/integration/api/iam/project_assignment.spec.ts [hunk 4]
```diff
@@ -0,0 +1,452 @@
...
+              name: testObjectID,
+              projects: []
+            }
+        }).then((resp) => {
+            expect(resp.status).to.equal(200);
+            expect(resp.body.team.projects).to.have.length(0);
+        });
+    });
+
+    it('assigning (unassigned) to authroized is allowed ' +
```
[the remark this answers is not in the document]
Comment (tylercloke) [reply e2e/cypress/integration/api/iam/project_assignment.spec.ts]: 😭😭😭

File: e2e/cypress/integration/api/iam/project_assignment.spec.ts [hunk 5]
```diff
@@ -0,0 +1,452 @@
...
+            },
+            failOnStatusCode: false
+        }).then((resp) => {
+            expect(resp.status).to.equal(200);
+            expect(resp.body.team.projects).to.have.length(1);
+            expect(resp.body.team.projects[0]).to.equal(authorizedProject.id);
+        });
+    });
+
+    it('not changing projects on update is allowed for authorizedOnlyProjReq ',  () => {
```
[the remark this answers is not in the document]
Comment (tylercloke) [reply e2e/cypress/integration/api/iam/project_assignment.spec.ts]: Everything in these tests have `iam:team:*` but varying degrees of `iam:project:assign`.

File: components/authz-service/server/v2/system_integration_test.go [hunk 6]
```diff
@@ -1,780 +0,0 @@
...
-
-	api_v2 "github.com/chef/automate/api/interservice/authz/v2"
-	"github.com/chef/automate/components/authz-service/testhelpers"
-)
-
-// In these tests, we assert that our default policies are in place, and do
-// their job as expected. For this, we setup a v2 server using a proper OPA
-// engine instance, and run a few queries.
-
-func TestIntegrationSystemPolicies(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (tylercloke) [reply components/authz-service/server/v2/system_integration_test.go]: I wanted to delete the `NewTestFramework` which I did in a follow up commit. Personally, I'd rather get away from having any integration tests that are half mocking things in the cereal world, even if that means a bit of extra conversion code. Down for whatever, that was just my reasoning.

File: e2e/cypress/integration/api/iam/iam_integration.spec.ts [hunk 7]
```diff
@@ -148,6 +148,23 @@ describeIfIAMV2p1('assigning projects', () => {
       });
     });
 
+    it(`when creating ${iamResources} without projects is allowed`,  () => {
```
Comment (bcmdarroch) [inline e2e/cypress/integration/api/iam/iam_integration.spec.ts]: in keeping with the other test names, I'd include the expected behavior:
`when creating ${iamResources} without projects is allowed, successfully creates unassigned ${iamResource}`

File: e2e/cypress/integration/api/iam/iam_integration.spec.ts [hunk 8]
```diff
@@ -214,6 +231,22 @@ describeIfIAMV2p1('assigning projects', () => {
       });
     });
 
+    it(`when creating ${iamResources} without any projects ` +
```
Comment (bcmdarroch) [inline e2e/cypress/integration/api/iam/iam_integration.spec.ts]: `when creating ${iamResources} without any projects and project assignment permission is missing for (unassigned), fails to create unassigned ${iamResource}`

File: e2e/cypress/integration/api/iam/iam_integration.spec.ts [hunk 9]
```diff
@@ -344,6 +377,42 @@ describeIfIAMV2p1('assigning projects', () => {
       });
     });
 
+    // (TC) We do allow users to remove their project, even if results in unassigned and
+    // they don't have iam:project:assign perms on unassigned (special case).
+    it('assigning authorized to (unassigned) is allowed ' +
```
Comment (bcmdarroch) [inline e2e/cypress/integration/api/iam/iam_integration.spec.ts]: I think this tests the same: https://github.com/chef/automate/blob/f1a7c43f5874872b9b9876aaf8d5017961f40119/e2e/cypress/integration/api/iam/iam_integration.spec.ts#L463-L476

**Pull request merged**